add new endpoint /connectivity to check the network status

maxl2287 opened the pull request:
This PR will bring a new endpoint called "/connectivity" to get information about the connectivity-status of the device.

akoshunyadi commented:
I think in this PR we should focus on the corresponding issue it means only add the new endpoint connectivity. 
Extending the event-subscription should be an additional issue+PR.

About connectivity status: I think we should consider the 2 types of connectivity. The question is, how to model it. 
One possibility could be without input parameter, and the response is: CONNECTED_DATA, CONNECTED_DATA_SMS, CONNECTED_SMS, NOT_CONNECTED ?

Another alternative is to add an input parameter with reachability type. 

3rd alternative is to add another endpoint.

I think I would prefer the first one, and then let see what the community says.

maxl2287 commented:
> One possibility could be without input parameter, and the response is: CONNECTED_DATA, CONNECTED_DATA_SMS, CONNECTED_SMS, NOT_CONNECTED ?

What do you mean by "without input parameter"?

Without the information about a device, it's not possible to identify which device is meant?


**Current json:** 

```json
{
  "device": {
    "phoneNumber": "123456789",
    "networkAccessIdentifier": "123456789@domain.com",
    "ipv4Address": {
      "publicAddress": "84.125.93.10",
      "publicPort": 59765
    },
    "ipv6Address": "2001:db8:85a3:8d3:1319:8a2e:370:7344"
  }
}
```

akoshunyadi commented:
yes, sorry, I meant without connectivity related input parameter 🙂 Sure, we need the device.

maxl2287 committed:
feat[#34]: add new endpoint /connectivity to check the network status of the device

akoshunyadi commented on code/API_definitions/device-status.yaml:
```diff
@@ -344,6 +385,29 @@ components:
+      required:
+        - connected
+      properties:
+        connected:
```
Maybe connectionStatus or connectivityStatus would be better, it's strange otherwise when connected=NOT_CONNECTED 🙂

akoshunyadi reviewed (commented):
I would also extend the scopes for 3 legged auth. Actually they are not used even for roaming...

maxl2287 committed:
feat: updated three_legged device-status-connectivity-read and response parameter

akoshunyadi commented on code/API_definitions/device-status.yaml:
```diff
@@ -42,6 +45,45 @@ tags:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ConnectivityStatusResponse'
+              examples:
```
Now the example doesn't fit to the response object

maxl2287 committed:
fix: updated examples

maxl2287 committed:
fix: remove DATA_SMS

akoshunyadi committed:
Create MoM-2023-09-13.md

akoshunyadi committed:
Update MoM-2023-09-13.md

eric-murray committed:
Merge pull request #73 from akoshunyadi/MoM-2023-09-13

MoM-2023-09-13

akoshunyadi committed:
Create MoM-2023-09-27.md

bigludo7 committed:
Update device-status.yaml

Aligned with CloudEvents format

akoshunyadi committed:
Update MoM-2023-09-27.md

akoshunyadi committed:
Update MoM-2023-09-27.md

bigludo7 committed:
Update device-status.yaml

Align with DG

akoshunyadi committed:
Create MoM-2023-10-11.md

bigludo7 committed:
Update device-status.yaml

Changed to 
content:
                  application/cloudevents+json

akoshunyadi committed:
Update MoM-2023-10-11.md

bigludo7 committed:
Update device-status.yaml

Took into account  team review requests

eric-murray committed:
Merge pull request #77 from akoshunyadi/MoM-2023-10-11

MoM-2023-10-11

bigludo7 committed:
Update device-status.yaml

Took into account @akoshunyadi review comments.

bigludo7 committed:
Merge pull request #76 from akoshunyadi/MoM-2023-09-27

MoM-2023-09-27

bigludo7 committed:
Update device-status.yaml

Updated following @sachinvodafone and @fernandopradocabrillo  review

bigludo7 committed:
Update device-status.yaml

Fixed typo for error 500 (Thanks @sachinvodafone )

bigludo7 committed:
Update device-status.yaml

Added 202 for GET /subscriptions/{subscriptionId} as suggested by @sachinvodafone

bigludo7 committed:
Update device-status.yaml

Remove 202 for GET following @fernandopradocabrillo  comment.
Rename scope Following @akoshunyadi comment

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Update code/API_definitions/device-status.yaml

Co-authored-by: Rafal Artych <121048129+rartych@users.noreply.github.com>

bigludo7 committed:
Merge pull request #75 from camaraproject/bigludo7-patch-4

Added Notification management aligned with CloudEvents format

maxl2287 committed:
feat: update based on CloudEvents

maxl2287 committed:
fix: typo

maxl2287 committed:
style: update headlines

maxl2287 committed:
style: revert whitespaces

maxl2287 committed:
fix: use 0.5.0-wip

bigludo7 committed:
Update device-status.yaml

Hello @maxl2287 
Thanks for the proposal.
Few cosmetic change and one code change proposal: The eventType org.camaraproject.device-status.v0.subscription-ends has not to be subscribed (see API description). This notification does not require dedicated subscription. But, the eventType could be send in the notification so I've introduced 2 definition of eventType: one with and one without subscription-ends. 

Hope it makes sense for you

maxl2287 committed:
feat: update SubscriptionEventType with allOf SubscriptionCreationEventType
